LearningJourney/Modules/Features/JLibrary/JLibraryTests/Sources/Scenes/ObjectivesLisst/Domain/ObjectivesListViewModelTests.swift

```diff
@@ -0,0 +1,152 @@
+        //Given
+        fetchObjectivesUseCaseSpy.objectiveToUse = .failure(.unauthorized)
+        
+        //When
+        sut.handleOnAppear()
+        
+        //Then
+        XCTAssertEqual(sut.objectives, .error(.notAuthenticated))
+    }
+    
+    func test_handleOnAppear_whenResultFailsWithUnknownError_shouldHaveUnknownError() {
+        //Given
+        fetchObjectivesUseCaseSpy.objectiveToUse = .failure(.unknown)
+        
+        //When
+        sut.handleOnAppear()
+        
+        //Then
+        XCTAssertEqual(sut.objectives, .error(.unknown({})))
+    }
+    
+    func test_handleOnAppear_whenResultFailsWithApiError_shouldHaveUnknownError() {
+        //Given
+        fetchObjectivesUseCaseSpy.objectiveToUse = .failure(.api(.notAllowed))
+        
+        //When
+        sut.handleOnAppear()
+        
+        //Then
+        XCTAssertEqual(sut.objectives, .error(.unknown({})))
+    }
+    
+    func test_handleOnAppear_whenResultFailsWithParsingError_shouldHaveUnknownError() {
+        //Given
+        fetchObjectivesUseCaseSpy.objectiveToUse = .failure(
+            .parsing(
+                .invalidData(DummyError.dummy)
+            )
+        )
+
+        //When
+        sut.handleOnAppear()
+
+        //Then
+        XCTAssertEqual(sut.objectives, .error(.unknown({})))
+    }
+    
+    // MARK: - handleDidLearnToggled
+    
+    func test_handleDidLearnToggled_shouldPassCorrectObjective() {
+        //Given
+        let firstLearningObjective = createLearningObjective(id: 1)
+        let secondLearningObjective = createLearningObjective(id: 2)
+        
+        toggleLearnUseCaseSpy.resultToUse = .success(secondLearningObjective)
+        
+        sut.objectives = .result([
+            LibraryViewModelState.result(firstLearningObjective),
+            LibraryViewModelState.result(secondLearningObjective)
+        ])
+        
+        //When
+        sut.handleDidLearnToggled(
+            objective: LibraryViewModelState.result(secondLearningObjective)
+        )
+        
+        //Then
+        XCTAssertEqual(toggleLearnUseCaseSpy.objectivePassed?.id, 2)
+    }
+    
+    func test_handleDidLearnToggled_whenObjectivesArrayIsEmpty_shouldNotCallExecute() {
+        //Given
+        let learningObjective = createLearningObjective(id: 1)
+        toggleLearnUseCaseSpy.resultToUse = .success(learningObjective)
+        sut.objectives = .result([])
+                
+        //When
+        sut.handleDidLearnToggled(
+            objective: LibraryViewModelState.result(learningObjective)
+        )
+        
+        //Then
+        XCTAssertFalse(toggleLearnUseCaseSpy.executeCalled)
+    }
+    
+    // MARK: - Helper functions
+    
+    private func createLearningObjective(id: Int) -> LearningObjective {
+        LearningObjective(
+            id: id,
+            code: "code",
+            description: "description",
+            isCore: Bool.random(),
+            isComplete: Bool.random()
+        )
+    }
```

Comment:
Consider creating a `fixture` for this. It makes it way easier to instantiate objects in our tests, since we can just use `.fixture()` and modify the parameters we need modified
```suggestion
extension LearningObjective {
	  static func fixture(
	  	id: Int = .init(),
	  	code: String = .init(),
	  	description: String = .init(),
	  	isCore: Bool = .random(),
	  	isComplete: Bool = .random()
	  ) -> LearningObjective  {
		    LearningObjective(
			      id: id,
			      code: "code",
			      description: "description",
			      isCore: Bool.random(),
			      isComplete: Bool.random()
		
		)
	  }
}
```